dkg/frost/instance.go

```diff
@@ -0,0 +1,244 @@
+package frost
+
+import (
+	"math/rand"
+
+	"github.com/bloxapp/ssv-spec/dkg"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"github.com/coinbase/kryptology/pkg/dkg/frost"
+	ecies "github.com/ecies/go/v2"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/pkg/errors"
+)
+
+var (
+	thisCurve  = curves.BLS12381G1()
+	domainType = types.PrimusTestnet
+	sigType    = types.DKGSignatureType
+)
+
+func init() {
+	types.InitBLS()
+}
+
+// Instance is a single FROST DKG instance. It implements Start to init DKG
+// parameters and broadcast session key, and ProcessMsg to process 2 rounds of
+// frost, blame and keygen output messages
+type Instance struct {
+	state  *State
+	config *Config
+
+	instanceParams InstanceParams
+}
+
+// New creates a new protocol instance for new keygen
+// TODO: func New(network dkg.Network, signer types.DKGSigner, storage dkg.Storage, config ProtocolConfig) {}
+func New(
+	network dkg.Network,
+	operatorID types.OperatorID,
+	requestID dkg.RequestID,
+	signer types.DKGSigner,
+	storage dkg.Storage,
+	init *dkg.Init,
+) dkg.Protocol {
+
+	instanceParams := InstanceParams{
+		identifier: requestID,
+		threshold:  uint32(init.Threshold),
+		operatorID: operatorID,
+		operators:  types.OperatorList(init.OperatorIDs).ToUint32List(),
+	}
+	return newProtocol(network, signer, storage, instanceParams)
+}
+
+// NewResharing creates a new protocol instance for resharing
+// Temporary, TODO: Remove and use interface with Reshare
+func NewResharing(
+	network dkg.Network,
+	operatorID types.OperatorID,
+	requestID dkg.RequestID,
+	signer types.DKGSigner,
+	storage dkg.Storage,
+	operatorsOld []types.OperatorID,
+	init *dkg.Reshare,
+	output *dkg.KeyGenOutput,
+) dkg.Protocol {
```

Comment:
Since we already create `Config` and `InstanceParams`, we don't need these two `New` functions anymore. We can use init/reshare to create `InstanceParams` and the `New` will just require `Config` and `InstanceParams`.

Reply:
Agree, will add this change